### PR TITLE
feat: add nprogress bar for page transition indication

### LINKED ISSUE #1639: Feature: add missing loading animations to page transitions
### Type of feature

🍕 Feature

### Current behavior

Right now we only show a loading splash screen when users click on the opensauced logo:

Kapture 2023-08-29 at 17 59 46

If users want to navigate the rest of the site (which can be a bit slow at times) there's no indication that the page is loading. 

### Suggested solution

We could really use some affordance to make it clear to the user that when they click on a link to navigate to somewhere else in the page, that page is loading. Otherwise it feels like you click on stuff but nothing happens.

As much as I like the pizza splash screen, I think it does remove the user from context and adds some cognitive load, making it feel like the page takes longer to load than it actually does.

A suggested solution would be to have a loading animation like Github's -- a loading bar right below the url box (see the blue loading indicator): 

Kapture 2023-08-29 at 17 54 01

### Example

(pls ignore weird background texture of the gif, focus on orange bar at the top)

```
height: 4px 
background-color: OpenSauced primary (#FF5402)
```

Kapture 2023-08-29 at 18 15 36

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR adds page transition loading indication to improve users navigation experience

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #1639 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/62995161/68796353-c002-4586-b16a-441b3d80c914

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -142,7 +143,7 @@ function MyApp({ Component, pageProps }: ComponentWithPageLayout) {
           provider: localStorageProvider,
         }}
       >
-        {/* <Toaster position="top-center" /> */}
+        <NextNProgress color="#f97316" height={4} />
```
Comment (nickytonline) [inline pages/_app.tsx:146]: Where is the orange color coming from? I searched the project for `#f97316` and it's no where to be found.

Do we want to used the OpenSauced orange, `hsla(19, 100%, 50%, 1)`? 

https://github.com/brainconnect93/open-sauced-app/blob/d2e73a9c29d06c949cc5a81ca82b6d29fbf755cd/tailwind.config.js#L36

Comment (OgDev-01) [reply pages/_app.tsx:150]: > Do we want to used the OpenSauced orange, hsla(19, 100%, 50%, 1)?

Yes!... will update to the hsl version👍

**Comment by isabensusan** [comment]
This is so much better already @OgDev-01 !!! Thanks!

I have two comments:
1. when clicking on the "OpenSauced" logo there's a loading page with the pizza icon, and the loading behavior is a bit weird.  It's like it shows the loading bar, but then it goes to the pizza page and then back to the feed with the loading bar again. Can we get rid of the pizza page altogether? It makes the experience feel sluggish

Rewatch GIF Recording - 2023-11-03 at 12 32 16 PM


2. I think the bar by itself looks great, can we remove the little loading wheel that show up on the top right corner?
Rewatch GIF Recording - 2023-11-03 at 12 36 40 PM

**Comment by OgDev-01** [comment]
> when clicking on the "OpenSauced" logo there's a loading page with the pizza icon, and the loading behavior is a bit weird. It's like it shows the loading bar, but then it goes to the pizza page and then back to the feed with the loading bar again. Can we get rid of the pizza page altogether? It makes the experience feel sluggish

Great feedback, @isabensusan... For the loading screen that shows up whenever you click on the sauced logo, I think this is currently something we have on the plate to work on, as it is related to the approach with redirecting users to the explore page. @nickytonline  will be fixing this sometime soon ⚡️.

> I think the bar by itself looks great. Can we remove the little loading wheel that show up on the top right corner?

I'll update to remove this

**Comment by OgDev-01** [comment]
The spinner is removed now @isabensusan ✨
https://github.com/open-sauced/app/assets/62995161/d5b2ee91-ef53-4c4e-a81c-13025ed78d9a

**Comment by nickytonline** [comment]
> > when clicking on the "OpenSauced" logo there's a loading page with the pizza icon, and the loading behavior is a bit weird. It's like it shows the loading bar, but then it goes to the pizza page and then back to the feed with the loading bar again. Can we get rid of the pizza page altogether? It makes the experience feel sluggish
> 
> Great feedback, @isabensusan... For the loading screen that shows up whenever you click on the sauced logo, I think this is currently something we have on the plate to work on, as it is related to the approach with redirecting users to the explore page. @nickytonline will be fixing this sometime soon ⚡️.
> 
> > I think the bar by itself looks great. Can we remove the little loading wheel that show up on the top right corner?
> 
> I'll update to remove this

Yeah the pizza pulsating animation will probably disappear in the near future. We’re doing redirects on the client-side that could be done in middleware instead.

**Review by bdougie — APPROVED** [review]
This is a nice addition. Clicking the Insights pages is still quite slow, but now I can see it working.

**Review by isabensusan — APPROVED** [review]

**Review by nickytonline — APPROVED** [review]
🚀

**Review by jpmcb — APPROVED** [review]
🚀

**Pull request merged**